lib/Doctrine/DBAL/Driver/Statement.php

```diff
@@ -23,8 +23,8 @@ interface Statement extends ResultStatement
      *                          this will be a parameter name of the form :name. For a prepared statement
      *                          using question mark placeholders, this will be the 1-indexed position of the parameter.
      * @param mixed      $value The value to bind to the parameter.
-     * @param int        $type  Explicit data type for the parameter using the {@link ParameterType}
-     *                          constants.
+     * @param mixed      $type  Explicit data type for the parameter using the {@link ParameterType}
```

Comment:
This change is invalid. All implementations of this interface must accept integers. A specific implementation (`Doctrine\DBAL\Statement`) accepts other types but they are not part of the interface.

Reply:
Thanks @morozov - is the documentation incorrect then on `Doctrine\DBAL\Connection::prepare`? FWIW changing the documentation on `prepare` to return a `Doctrine\DBAL\Statement` fixes the issue for us.

Reply:
It looks like it's already fixed in https://github.com/doctrine/dbal/commit/15d9be0b3c87e05f428f8c2beb1fb9555e7b75bf but not yet released.

Reply:
Are there plans to create a release in the near future?

Reply:
We're waiting for #3980 to be resolved in order to release `2.10.3`, `2.11.0`, and `3.0.0`. We can release `2.10.3` earlier but there's no reason for that right now.